### DISCUSSION TITLE
Making sure only one operator is active

### DISCUSSION THREAD

**Discussion by ghost** [operator-framework/java-operator-sdk#1092] [Q&A]
During development, I've now had it happen multiple times that I had my operator running deployed in the cluster as well as in the IDE, or by pilot error having the operator run both as a Run and a Debug configuration in IDEA. This leads to really confusing behavior :-)

Is there an easy and or recommended way to somehow mark the resource with a unique ID of the running operator, so that the second operator can see that the resource is already managed by another instance, and ignores it?

I've looked at ownerReferences[].controller, but I'm not sure what to put in there to identify the running controller instance.

**Comment by csviri** [comment]
I don't think there is an easy way to handle that. At least I cannot think of one :)  
Only one operators should handle the resources for sure. We cannot really even identify it by some name and mark it, even the pod name changes all the time. So I don't think this is currently solved by any of the frameworks.

**Reply by metacosm** [reply]
The ideas explored in https://github.com/java-operator-sdk/java-operator-sdk/issues/877 could be useful for that purpose, though we probably don't want to go there for this specific use case :)

**Reply by metacosm** [reply]
Another more practical option would be for your operator to use the commit id from `ConfigurationService.getVersion` and add a label to your custom resources whenever they're handled. Then your reconciler would check the label and if it doesn't match its own version, then it wouldn't reconcile the resource (or something along those lines).

**Reply by ghost** [reply]
I was thinking along these lines. Maybe have the operator instance create a random unique ID at startup, and mark the resource with that. If it gets handed a resource later that has a different id, ignore it. Of course, that means that resources could get abandoned when the operator instance gets shut down and fails to un-mark the resources it had claimed for itself.

it's not an urgent problem, just annoying as a dev when you manage to confuse yourself that way.
